code/datums/ai/movement/_ai_movement.dm

```diff
@@ -21,3 +21,40 @@
 	controller.pathing_attempts++
 	if(controller.pathing_attempts >= max_pathing_attempts)
 		controller.CancelActions()
+
+//Instead of copypasting/subtype and rewrite all the movement checks in pre_move() you can use this
+/datum/ai_movement/proc/allowed_to_move(datum/move_loop/source)
+	SIGNAL_HANDLER
```

Comment:
As far as I can tell, allowed_to_move is just a standard proc, do you have a specific reason to have it registered as a SIGNAL_HANDLER or is it because it was copied from `/datum/ai_movement/[whatever]/proc/pre_move`? SIGNAL_HANDLER is specifically used for procs that are triggered by signals

Reply:
SIGNAL_HANDLER should've been on pre_move()